pkg/loadbalancerservice/controller.go

```diff
@@ -154,7 +154,7 @@ func (c *LoadbalancerServiceController) updateServiceStatus(key string) error {
 		klog.Infof("Service %s does not exist anymore", key)
 	} else {
 		svc := obj.(*corev1.Service)
-		if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		if svc.Spec.Type != corev1.ServiceTypeLoadBalancer || (svc.Spec.LoadBalancerClass != nil && *svc.Spec.LoadBalancerClass != "microshift-internal") {
```

Comment:
While the controller is too open now (thanks for catching that!), this might be a bit too restrictive. This controller acts not just on MicroShift-owned resources, but also on any `LoadBalancer` `service` owned by applications.
By requiring the `loadBalancerClass` to match an expected value, MicroShift would be imposing that need to already deployed services and they would fail to pick up IPs again. This would happen in an upgrade, for example. Also, requiring this value in application's manifests would break compatibility with OpenShift and would require specific configuration for apps to run on MicroShift.
Instead of checking a specific value of the `loadBalancerClass`, I would check for emptiness instead.
From the [docs](https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class):
```
If you specify .spec.loadBalancerClass, it is assumed that a load balancer implementation that matches the specified class is watching for Services. Any default load balancer implementation (for example, the one provided by the cloud provider) will ignore Services that have this field set.
```

I would drop the second part of the new check and that should work in all cases.

Reply:
Thanks for reviewing my request! Yep, that makes perfect sense! I updated the PR with only the check for nil now 😄